hasattachments.go

```diff
@@ -0,0 +1,6 @@
+package allure
+
+type hasAttachments interface {
```

Comment:
Would it make sense to put all the hasSomething interfaces in the existing something.go files?
It looks messy to have all those single hasSomething.go files

Reply:
This is addressed in [Variadic functions PR](https://github.com/dailymotion/allure-go/pull/22).

Reply:
Merged master with this branch, as a result moved interfaces to a separate file.